lib/beamchmark/suite/measurements/cpu_info.ex

```diff
@@ -79,17 +57,42 @@ defmodule Beamchmark.Suite.Measurements.CpuInfo do
         acc + map.average_all_cores
       end) / length(cpu_usage_unstable_list)
 
+    sum_by_core =
+      Enum.reduce(cpu_usage_unstable_list, %{}, fn %{cpu_usage: cpu_usage}, sum_cores_acc ->
+        cpu_usage |> reduce_cpu_usage(sum_cores_acc)
+      end)
+
+    number_of_snapshots = length(cpu_usage_unstable_list)
+
+    average_by_core =
+      Enum.reduce(sum_by_core, %{}, fn {core_id, value}, acc ->
+        Map.put(acc, core_id, Float.round(value / number_of_snapshots, 2))
+      end)
+
     %__MODULE__{
       cpu_snapshots: cpu_usage_unstable_list,
+      average_by_core: average_by_core,
       average_all: average_all
     }
   end
 
   @spec diff(t(), t()) :: t()
   def diff(base, new) do
+    average_by_core_diff =
+      Enum.reduce(new.average_by_core, %{}, fn {core_id, value}, acc ->
+        Map.put(acc, core_id, value - Map.get(base.average_by_core, core_id))
+      end)
+
     %__MODULE__{
       cpu_snapshots: base.cpu_snapshots,
-      average_all: new.average_all - base.average_all
+      average_all: new.average_all - base.average_all,
```

Comment:
I agree that it is the best place to calculate `percent_difference`, but:
* We need to have old number as well (or usual difference), because we are showing end user not only percent difference, but value difference as well. (As it is done in `format_numbers` initially)
* We could calculate backwards the old value using percent difference, but then this change won't make sens.
Can you please specify how you would see those changes?

Reply:
Ok, so let's keep current version. I will create an issue to make API more consistent